src/Piral.Blazor.Tools/tasks/ManagePiletTask.cs

```diff
@@ -191,24 +191,24 @@ private int GetNpmVersion()
             return 0;
         }
 
-        private void CopyConfigurationFiles()
+        private void InitializeNpmConfiguration()
         {
-            var configurationFiles = new[] { ".npmrc" };
+            var configFileName = ".npmrc";
             var source = ConfigDir;
             var target = ProjectDir;
 
-            Log.LogMessage($"Copying config files from '{source}'...");
+            var existingConfigFile = Path.Combine(source, configFileName);
 
-            foreach (var configurationFile in configurationFiles)
+            if (File.Exists(existingConfigFile))
             {
-                var sourceFile = Path.Combine(source, configurationFile);
-
-                if (File.Exists(sourceFile))
-                {
-                    var targetFile = Path.Combine(target, configurationFile);
-                    Log.LogMessage($"Copying file '{configurationFile}' from '{sourceFile}'.");
-                    File.Copy(sourceFile, targetFile, true);
-                }
+                Log.LogWarning($"Ignoring NpmRegistry property! Copying file '{configFileName}' from '{existingConfigFile}'.");
```

Comment:
Why do we warn here? Is something in the NpmRegistry that is about to be ignored? We don't know here / there is no check.

Reply:
We are showing warning so user knows that whatever is specified under NpmRegistry won't matter as custom `.npmrc` has higher priority and it's being used over NpmRegistry property.

Reply:
Yeah but a warning is a sign that something is not right. We should only print a warning in case both are specified. Otherwise I'd expect just an info log (Taking "X", e.g., "Taking entry from NpmRegistry" or "Taking custom .npmrc").

Reply:
Okay, I'll move that to be an info log. I'm unsure on how to check if NpmRegistry property has been user specified or if we are using the default one. Do you have an approach we could use for that? All I can think of is to provide "DefaultNpmRegistry" parameter which will not be Property that is replaceable by user, and use that to check if provided NpmRegistry is different than the default one.

Reply:
Well I'd just trim the string and see if anything is in there. Additionally we could just if its a valid URL, however, if its anything I'd just take it. In case its really invalid (or just the default one) it does not matter imho.

Reply:
I'm kinda lost now. We are unable to specify empty NpmRegistry, as it will always use the default one, so it will never take custom `.npmrc` file.

What if we just leave it as it is right now and just update the log? If there's `.npmrc`, it will copy and use that, and if there's no `.npmrc` file, it will use whatever is specified under NpmRegistry (default or user specified).

Reply:
Yes let's do it like this.

Reply:
I have updated log messages.

What about in case info file (`.blazorrc`) exists, should we do proposed check you mentioned with checksum/registry value or just execute the same method again so in case registry has changed, it will update `.npmrc` based on the same logic we have right now in the `InitializeNpmConfiguration` method?